### PR TITLE
First pass at the REST endpoints for roles.

### PR DESCRIPTION
**Pull request by theothertomelliott**
Couple of outstanding questions:
* Should we error out for bundles and permissions that currently don't exists?
* Should the role permissions endpoints account for bundle versions?

### CODE REVIEW

File: dataaccess/interface.go [hunk 1]
```diff
@@ -66,6 +66,7 @@ type DataAccess interface {
 	RoleCreate(ctx context.Context, rolename string) error
 	RoleDelete(ctx context.Context, rolename string) error
 	RoleGet(ctx context.Context, rolename string) (rest.Role, error)
+	RoleExists(ctx context.Context, rolename string) (bool, error)
```
Comment (clockworksoul) [inline dataaccess/interface.go:69]: Good catch.

File: service/group-handlers_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+import (
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/getgort/gort/data/rest"
+)
+
+func TestGrantGroupRole(t *testing.T) {
```
Comment (clockworksoul) [inline service/group-handlers_test.go:12]: Tests, yay!!

**Review by clockworksoul — CHANGES_REQUESTED** [review]
The only nit is the `jsonRequest` function name, but otherwise everything looks good!

**Comment by clockworksoul** [comment]
Oh, also, while you're in there, would you mind incrementing the version to `0.7.5-dev.0` ?

File: service/service_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	"io"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gorilla/mux"
+
+	"github.com/getgort/gort/dataaccess/memory"
+)
+
+func jsonRequest(t *testing.T, router *mux.Router, method string, target string, body interface{}, out interface{}) int {
```
[the remark this answers is not in the document]
Comment (theothertomelliott) [reply service/service_test.go]: Refactored as discussed.

**Pull request merged**